common/azure-common/src/main/java/com/azure/common/implementation/serializer/HttpResponseBodyDecoder.java

```diff
@@ -219,6 +224,31 @@ private static Type constructWireResponseType(Type resultType, Type wireType) {
         return wireResponseType;
     }
 
+    /**
+     * Deserializes a response body as a Page<T> given that {@param wireType} is either:
+     * 1. A type that implements the interface
+     * 2. Is of {@link Page}
+     *
+     * @param value The string to deserialize
+     * @param resultType The type T, of the page contents.
+     * @param wireType The {@link Type} that either is, or implements {@link Page}
+     * @param serializer The serializer used to deserialize the value.
+     * @param encoding Encoding used to deserialize string
+     * @return An object representing an instance of {@param wireType}
+     * @throws IOException if the serializer is unable to deserialize the value.
+     */
+    private static Object deserializePage(String value, Type resultType, Type wireType, SerializerAdapter serializer, SerializerEncoding encoding) throws IOException {
+        final Type wireResponseType;
+
+        if (wireType != Page.class) {
+            wireResponseType = wireType;
+        } else {
+            wireResponseType = TypeUtil.createParameterizedType(ItemPage.class, resultType);
```

Comment:
nit; A comment would be helpful, something like:

If the type is 'Page' interface [i.e. `@ReturnValueWireType(Page.class)`], we will use the 'ItemPage' class instead.

I would use +ve case first to make it a little more readable, if that ok with you?

```java
if (wireType == Page.class) {
   // If the type is the 'Page' interface [i.e. `@ReturnValueWireType(Page.class)`], we will use the 'ItemPage' class instead.
    wireResponseType = TypeUtil.createParameterizedType(ItemPage.class, resultType);
} else {
    wireResponseType = wireType;
}
```